### PR TITLE
dispersive pulse reduce number of calcualtions

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- reduce the number of trigonometric function calls to one per frequency
- reduce the number of multiplications

### CODE REVIEW

**Review by ikbuibui — APPROVED** [review]

**Review by fabidie — APPROVED** [review]
smart move!

File: include/picongpu/fields/incidentField/profiles/DispersivePulse.hpp [hunk 1]
```diff
@@ -331,17 +331,19 @@ namespace picongpu
                             // electric field in time-domain
                             float_X E_t = 0.0_X;
 
+                            /* E_t = sum(2 * amp(k) * (cos(phi) * cos(omegaT(k)) + sin(phi) * sin(omegaT(k)))) / dt) ==
```
Comment (steindev) [inline include/picongpu/fields/incidentField/profiles/DispersivePulse.hpp:334]: The `)` after `dt` shouldn't be there, no?

File: include/picongpu/fields/incidentField/profiles/DispersivePulse.hpp [hunk 2]
```diff
@@ -331,17 +331,19 @@ namespace picongpu
...
+                             * E_t = 2 * sum(amp(k) * (cos(phi) * cos(omegaT(k)) + sin(phi) * sin(omegaT(k)))) / dt ==
+                             * E_t = 2 * sum(amp(k) * (cos(phi - omegaT(k)))) / dt
```
Comment (steindev) [inline include/picongpu/fields/incidentField/profiles/DispersivePulse.hpp:336]: the parenthesis around the cos term is not required.

File: include/picongpu/fields/incidentField/profiles/DispersivePulse.hpp (hunk 1 above)
Comment (psychocoderHPC) [reply include/picongpu/fields/incidentField/profiles/DispersivePulse.hpp:334]: it should be it is the counter breaket from `sum(`

**Pull request merged**